### ISSUE TITLE
Why The HashBang is necessary (sometimes)

### ISSUE THREAD

**Issue by leeatkinson** [browserstate/history.js#73]
Reading the https://github.com/balupton/History.js/wiki/Intelligent-State-Handling, it struck me that, for the following scenario, hashbangs are necessary:
1. The user is using a browser that does not support history.pushState() etc.
2. The user visits an AJAX-enabled website and selects a link that updates part of the page via AJAX.
3. The website, being a 'good' AJAX application, updates the window location to represent the current state. - http://mysite.com/section for 'pushState' browsers, http://mysite.com/#section for others.
4. Whilst in the site, the user decide to copy-and-paste the URL from the address bar and put that link on a second website.
5. A search engine crawler indexes the second website and finds the link above. As it has a hash in the URI, the search engine will index http://mysite.com. It is possible that the crawler may eventually crawl to http://mysite.com/section via links from the home page, but it is less likely to reach it than if it indexed it directly.

If the website had used hashbangs for non-'pushState' browsers, Google's crawler would have re-wrote http://mysite.com/#!section to http://mysite.com/?_escaped_fragment_=section and then indexed that.

In summary, I do think that hasbangs do have their use, but for this very specific (but important) use. I agree that most of the time, it is over-used.

Regards
